assets/lokomotive-kubernetes/dns/route53/main.tf

```diff
@@ -1,27 +1,36 @@
-variable "entries" {
-  type = list(
-    object({
-      name    = string
-      type    = string
-      ttl     = number
-      records = list(string)
-    })
-  )
+provider "aws" {
```

Comment:
Hm, I don't think modules should be instantiating the providers.

Reply:
The module didn't run for me without this since the AWS provider requires a region and I need to instantiate the provider to specify the region AFAIK. Do you have an alternative?

Reply:
The instantiating code should also instantiate the provider as far as I know. So this block should be conditionally added together with module instantiation in Packet's Terraform template I guess. Something like:

```tf
provider "aws" {
  ...
}

module "dns" {
  source = "..."
}
```

I'm not sure how big deal that is though. Perhaps we will find out, when porting those changes to AWS for example.

Reply:
@invidian +1. The module should use the [providers meta-argument] if a custom one is needed. If not, it seems cleaner and according to terraform best practices to define it in the top level terraform file.

Should we create an issue to track this? :)

[link]: https://www.terraform.io/docs/configuration/modules.html#other-meta-arguments

Reply:
Created #539.